dart-parser/src/parser/comment.rs

```diff
@@ -0,0 +1,65 @@
+use std::str;
+
+use nom::{
+    branch::alt,
+    bytes::complete::{is_not, tag},
+    combinator::{cut, eof, recognize},
+    multi::fold_many0,
+    sequence::{preceded, terminated, tuple},
+    IResult, Parser,
+};
+
+use crate::{dart::comment::Comment, parser::common::*};
+
+pub fn comments(s: &str) -> IResult<&str, Comment> {
+    alt((
+        comment.map(Comment::SingleLine),
+        multi_line_comment.map(Comment::MultiLine),
+    ))(s)
+}
+
+fn comment(s: &str) -> IResult<&str, &str> {
+    recognize(tuple((tag("//"), is_not("\r\n"), alt((br, eof)))))(s)
+}
+
+fn multi_line_comment(s: &str) -> IResult<&str, &str> {
+    recognize(preceded(
+        tag("/*"),
+        cut(terminated(
+            recognize(fold_many0(
+                alt((is_not("/**/"), multi_line_comment)),
```

Comment:
`is_not("/**/")` is likely not what you want. Consider `not(alt((tag("/*"), tag("*/"))))`